### PR TITLE
feat: Gamify join and leave application

### PR DESCRIPTION
**Pull request by sanderelderman**
Functionality:
1. Added pivot entity between `application` and `user`
2. An endpoint for joining an application
3. An endpoint for leaving an application

Note: merge #279 before this PR

### CODE REVIEW

File: libs/gamify/applications/src/lib/models/application.model.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export class ApplicationModel {
```
Comment (mschilling) [inline libs/gamify/applications/src/lib/models/application.model.ts:1]: Any specific reason why you don't want to name this `Application`?

File: libs/gamify/applications/src/lib/models/application-user.model.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export class ApplicationUserModel {
```
Comment (mschilling) [inline libs/gamify/applications/src/lib/models/application-user.model.ts:1]: Any specific reason why you don't want to name this `ApplicationUser`?

**Review by mschilling — APPROVED** [review]
🚀

File: libs/gamify/applications/src/lib/models/application.model.ts (hunk 1 above)
Comment (sanderelderman) [reply libs/gamify/applications/src/lib/models/application.model.ts:1]: Nope, no particular reason

**Pull request merged**